### DISCUSSION TITLE
[Feature] Ability to change date and time format

### DISCUSSION THREAD

**Discussion by Arcturuss** [johanohly/AirTrail#73] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

Currently there is only one date and time format with no ability to change it: in the date picker, in the flight list, in the flight info popup on the map. Additionally, flight time is displayed in flight edit form in 12-hour format, even if I entered it in 24-hour format.

It would be nice to have option to select or customize date and time format.

![image](https://github.com/user-attachments/assets/236a4025-86dc-4a36-a7b8-8f07fe08b07a)
![image](https://github.com/user-attachments/assets/9bb15804-1e2d-45bb-bdfe-8c1d212f2160)

**Comment by johanohly** [comment]
Thank you for making a feature request!

Most places use the format that matches your browsers' preference (actually, it is your browser that formats it entirely).
The only exception to this is the time input (not date), which you correctly observed converts times to 24h regardless of browser preference. This is definitely an oversight on my end, and will be fixed in the next release.

Is there something else I am missing?

**Reply by Arcturuss** [reply]
While the browser locale can be good first guess for the date and time format, sometimes the preferred format is different and needs to be set explicitly.

But currently it's not consistent in the app. Just checked with different locale in the browser:
- date and time format in the flight list actually follows the browser locale
- date in the popup on the map does not
- time input in the add/edit flight form follows the locale (12/24h format)
- date picker does not

![image](https://github.com/user-attachments/assets/289a9a69-fed9-42c7-b190-7ac8484c6856)
![image](https://github.com/user-attachments/assets/ae632cf5-893d-42d5-bced-2ca034648624)
![image](https://github.com/user-attachments/assets/78f11312-805c-459a-9bf8-a0d4b61b59ea)

**Comment by johanohly** [comment]
The only missing place I could find was the map airport popups. They are localized in the next release. 
![image](https://github.com/user-attachments/assets/d846583b-30d9-4bf8-91cc-a9ef779e8329)
This part of the date picker is in fact localized, although using a slightly different method, as the "token orders" (mm, dd, yyyy) have to be hard-coded. This is handled by melt-ui.
